<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Background • slouch</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/united/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Background">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">slouch</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">2.1.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/background.html">Background</a>
    </li>
    <li>
      <a href="../articles/examples.html">Examples</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="http://github.com/kopperud/slouch">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Background</h1>
                        <h4 class="author">Jason Pienaar, Bjørn Tore Kopperud, Kjetil Lysne Voje, Thomas F. Hansen (in no particular order)</h4>
            
            <h4 class="date">2018-10-02</h4>
      
      <small class="dont-index">Source: <a href="http://github.com/kopperud/slouch/blob/master/vignettes/background.Rmd"><code>vignettes/background.Rmd</code></a></small>
      <div class="hidden name"><code>background.Rmd</code></div>

    </div>

    
    


<div id="comparative-methods-for-adaptive-hypotheses" class="section level1">
<h1 class="hasAnchor">
<a href="#comparative-methods-for-adaptive-hypotheses" class="anchor"></a>Comparative methods for adaptive hypotheses</h1>
<div id="background" class="section level2">
<h2 class="hasAnchor">
<a href="#background" class="anchor"></a>Background</h2>
<p>Comparing traits in groups of related species in different environments or time periods can provide powerful tests of adaptive hypotheses (e.g. Sober 2008; Hansen 2014). The distributions of trait values of extant species at the tips of a phylogeny (and fossil data or reconstructed ancestral states within the phylogeny if available) however, may reflect, adaptation, phylogenetic inertia, both of these, or none of these. We use the term phylogenetic inertia in the sense of resistance to adaptive change i.e. it comes about when related species inherit an inert trait from a common ancestor, which then lags adaptive change to new optima. Hence, if inertia is present, observed trait values may reflect both adaptive change as well as an influence of a maladapted ancestral trait. When testing an adaptive hypothesis with such data, the constraints provided by such resistance to change for current trait values as they evolve towards their optima need to be controlled for (and similarly, the effects of adaptation need to be controlled for when testing hypothesis of phylogenetic inertia).</p>
<p>Most contemporary comparative methodologies do not control for inertia in the sense defined above, but rather control for general effects of phylogeny. Their principal concern is to deal with statistical issues that arise from non-independent, correlated data that naturally arise from evolution along a branching phylogeny, and are usually used to remove these correlations before the actual analysis takes place. Phylogenetic effects, however, will also be present when traits adapt to variables that are themselves phylogenetically structured. Effects of adaptation should not be controlled for when studying adaptation and they need to be separated from inertia. Furthermore, most contemporary methods assume that traits evolve as a Brownian motion, a crucial assumption and one that provides the analytical techniques to remove the correlations (e.g. independent contrasts). Numerous studies however, have pointed out that comparative method that assumes a Brownian motion as the underlying evolutionary process are not suitable for studying adaptation towards optima (e.g. Hansen 2014). A Brownian motion is a stochastic process in which the expected mean value equals the common ancestral state at the base of the species phylogeny. There is no mechanism in the process that allows one to specify adaptation to or maintenance at specific optima. Even if the ancestral state is at the optimum and the optimum does not change through time, any deviations from the optimum relationship generated by the stochastic Brownian-motion process will be inherited and there is no mechanism that allows a trait in a species to return to its optimum value.</p>
<p>Hansen (1997), Butler &amp; King (2004), Hansen &amp; Orzack (2005) and Hansen <em>et al.</em> (2008) have argued that comparative methods based on an Ornstein-Uhlenbeck (OU) process are more suited to studying adaptive hypotheses using comparative data. The OU process, unlike a pure Brownian motion, consists of a deterministic component (which can be used to model adaptation of a trait towards an optimum) as well as a stochastic component (which models the random fluctuations of the trait as it evolves towards the optimum. If we take away the deterministic component of an OU process, we are left with a Brownian motion.</p>
</div>
<div id="phylogeny" class="section level2">
<h2 class="hasAnchor">
<a href="#phylogeny" class="anchor"></a>Phylogeny</h2>
<p>These methods require a rooted phylogeny with branch lengths. Both ultrametric and non-ultrametric trees can be used, but this will have implications for reliable estimation of ancestral states. Polytomies and non-branching edges are not a problem but each branch must have a length. The units of the lengths are arbitrary and can be given in time units or number of nucleotide substitutions. A lot of the power of this method comes from having accurate branch lengths so some care should be taken in time-calibrating the phylogeny.</p>
</div>
</div>
<div id="comparative-methods-built-around-an-ornstein-uhlenbeck-process" class="section level1">
<h1 class="hasAnchor">
<a href="#comparative-methods-built-around-an-ornstein-uhlenbeck-process" class="anchor"></a>Comparative methods built around an Ornstein-Uhlenbeck process</h1>
<p>The Ornstein-Uhlenbeck process describes stochastic evolution with a deterministic tendency to move towards a fixed state (Hansen &amp; Martins 1996). To model adaptive evolution in a comparative data set, we interpret this fixed state as a “primary optimum”, defined as the average fitness optimum that would be reached by a large number of independent species evolving for a sufficiently long time in a given niche to be free of any ancestral influence (Hansen 1997). The model assumes that primary optima exist at any point in time (on the phylogeny), and the idea is to test hypotheses about the effect of specified environmental variables on the primary optima. Currently, the software allows the primary optimum to be be modeled as a multiple regression on several continuous “random” variables, continuous “direct-effect” covariates, or as a one-way ANOVA on fixed, categorical variables mapped as regimes on the phylogenetic tree.</p>
<div id="the-optimum" class="section level2">
<h2 class="hasAnchor">
<a href="#the-optimum" class="anchor"></a>The optimum</h2>
<p>If we let <span class="math inline">\(\theta(x_1, x_2 \dots)\)</span>, be the primary optimum as a function of environmental variables, <span class="math inline">\(x_1, x_2 \dots\)</span>, then for the regression model, we have:</p>
<p><span class="math display">\[
\theta(x_1, x_2 \dots) = b_0 + b_1x_1 + b_2x_2 \dots,
\]</span></p>
<p>where <span class="math inline">\(b_0\)</span> is an intercept, and <span class="math inline">\(b_1, b_2 \dots\)</span> are regression slopes on the environmental variables. For the ANOVA model we let <span class="math inline">\(x_1, x_2 \dots\)</span> be indicator variables indicating different categorical states of the environment, and we write</p>
<p><span class="math display">\[
\theta(x_1, x_2 \dots) = \theta_1x_1 + \theta_2x_2 \dots,
\]</span> where <span class="math inline">\(\theta_1x_1, \theta_2x_2 \dots\)</span> are parameters describing the state of the primary optimum in the different environments or niches. The goal is to estimate the parameters <span class="math inline">\(b_i\)</span> or <span class="math inline">\(\theta_i\)</span> to test whether particular aspects of the environment have an effect on the optimum or not. Our adaptive hypotheses thus takes the form: Given that there is a single optimum, the species are adapted to variable <span class="math inline">\(x\)</span> if <span class="math inline">\(x\)</span> has an (important) effect on the optimum. As we will see, the methods also allow for assessments of how strongly the species tend to evolve towards the optimum, and thus to reject the base assumption of adaptation altogether.</p>
</div>
<div id="ornstein-uhlenbeck-process" class="section level2">
<h2 class="hasAnchor">
<a href="#ornstein-uhlenbeck-process" class="anchor"></a>Ornstein-Uhlenbeck process</h2>
<p>In both the fixed categorical and continuous random predictor cases, the evolution of the response trait, <span class="math inline">\(y\)</span>, can be represented by the stochastic differential Ornstein-Uhlenbeck process equation:</p>
<p><span class="math display">\[
dy = -\alpha(y - \theta)dt + \sigma_ydW_y,
\]</span></p>
<p>which is interpreted as follows: <span class="math inline">\(dy\)</span> is the change in <span class="math inline">\(y\)</span> over a time step <span class="math inline">\(dt\)</span>, <span class="math inline">\(\alpha\)</span> is a parameter measuring the rate of adaptation towards the optimum, <span class="math inline">\(dW_y\)</span> is a white-noise process having independent, normally-distributed random changes with mean zero and unit variance. The standard deviation of the random changes is given by <span class="math inline">\(\sigma_y\)</span> and <span class="math inline">\(\theta\)</span> are the fixed or random optima as defined above. Categorical predictor variables must be mapped on the phylogeny a priori, and the method treats these as fixed.</p>
</div>
<div id="parameterization" class="section level2">
<h2 class="hasAnchor">
<a href="#parameterization" class="anchor"></a>Parameterization</h2>
<p>We note that we will work with the transformed parameters:</p>
<p><span class="math display">\[
\begin{aligned}
t_{1/2} =&amp; \frac{\log(2)}{\alpha} \\
v_y =&amp; \frac{\sigma^2_y}{2\alpha},
\end{aligned}
\]</span> These are both easier to interpret than <span class="math inline">\(\alpha\)</span> and <span class="math inline">\(\sigma_y\)</span>. The phylogenetic half life, <span class="math inline">\(t_{1/2}\)</span>, is the time it takes for the expected trait value to move half the distance from the ancestral state to the primary optimum. The half life provides a useful metric to asses the strength of phylogenetic inertia as it is on the same linear scale as the phylogenetic branch lengths. If the half life is short relative to the total length of the phylogeny, it means adaptation to the primary optimum is rapid in expectation, and if the half life is long, it suggests the presence of inertia and that ancestral trait values influence the species’ observed trait values at the tips. Consequently, if the half life is large, we would expect the species traits to be poorly adapted to the primary niches. A half life of infinity corresponds to evolution governed by a Brownian-motion process in which there is no tendency to move towards the optimum, and a half life of zero corresponds to immediate adaptation. The parameter <span class="math inline">\(v_y\)</span> is the variance expected when the adaptive and stochastic forces are in stochastic equilibrium.</p>
</div>
<div id="random-continuous-predictor" class="section level2">
<h2 class="hasAnchor">
<a href="#random-continuous-predictor" class="anchor"></a>Random continuous predictor</h2>
<p>For the random predictors we have an additional stochastic differential Brownian-motion process to describe their evolution on a phylogeny:</p>
<p><span class="math display">\[
dx = \sigma_xdW_y,
\]</span></p>
<p>where <span class="math inline">\(\sigma_x\)</span> gives the standard deviation of the random changes (hence, the rate at which the predictors change). These dynamical equations can be solved for the different assumptions about the optimum.</p>
</div>
<div id="categorical-fixed-predictors-niches-regimes" class="section level2">
<h2 class="hasAnchor">
<a href="#categorical-fixed-predictors-niches-regimes" class="anchor"></a>Categorical fixed predictors (niches, regimes)</h2>
<p>With fixed, categorical predictor variables, the model predicts that the species data will be of the form:</p>
<p><span class="math display">\[
y_i = c_{0i}y_a + c_{1i}\theta_1 + c_{2i}\theta_2 \dots + r_i,
\]</span> where <span class="math inline">\(y_a\)</span> is a parameter describing the state of the trait at the root of the tree, the coefficients, <span class="math inline">\(c_{ji}\)</span> , are determined by the time on the phylogeny the species <span class="math inline">\(i\)</span> has spent in environment <span class="math inline">\(j\)</span>, such that more recent environments are weighted more than more ancient environments. The weights are determined by the rate of adaptation, <span class="math inline">\(\alpha\)</span>. More specifically, <span class="math inline">\(c_{ji}\)</span> is a sum over all time intervals in the past history of species <span class="math inline">\(i\)</span> where this species experienced environment <span class="math inline">\(j\)</span>, and each interval contributes a term <span class="math inline">\(e^{-\alpha t} - e^{-\alpha s}\)</span>, where, starting at the root with <span class="math inline">\(t_0\)</span> = 0, <span class="math inline">\(s\)</span> is the time when the species enters the environment, and <span class="math inline">\(t\)</span> is the end-time of the environment (see Hansen (1997) for formal derivation). The coefficient for the ancestral state <span class="math inline">\(y_a\)</span> is <span class="math inline">\(c_0 = e^{ -\alpha t}\)</span> , where <span class="math inline">\(t\)</span> is the time from the root to species <span class="math inline">\(i\)</span>. Note that if <span class="math inline">\(\alpha\)</span> is large, then only the most recent environment will contribute to the trait values of the species. Ancient environments only contribute to species’ trait values if the rate of adaptation is low. SLOUCH will automatically compute these coefficients given a mapping of niches on the tree. The residual terms, <span class="math inline">\(r_i\)</span>, are assumed to be normally distributed with mean zero and variances and covariances given later.</p>
</div>
<div id="direct-effect-predictors" class="section level2">
<h2 class="hasAnchor">
<a href="#direct-effect-predictors" class="anchor"></a>Direct-effect predictors</h2>
<p>For some biological systems (e.g. allometry) it may be more realistic to model the influence of <span class="math inline">\(x\)</span> on <span class="math inline">\(y\)</span> as direct, non-lagged process. In this case, we can expand the Ornstein-Uhlenbeck differential equation:</p>
<p><span class="math display">\[
dy = -\alpha(y - \theta)dt + bdx + \sigma_ydW_y,
\]</span> where changes in the predictor (<span class="math inline">\(dx\)</span>) has an immediate effect on changes in the response trait (<span class="math inline">\(dy\)</span>). The direct-effect model makes no assumptions of any phylogenetic process that gives rise to <span class="math inline">\(x\)</span>.</p>
</div>
<div id="example-expectation-with-random-covariates" class="section level2">
<h2 class="hasAnchor">
<a href="#example-expectation-with-random-covariates" class="anchor"></a>Example: expectation with random covariates</h2>
<p>For random, continuous predictor variables, evolving as if by a Brownian motion, the model predicts:</p>
<p><span class="math display">\[
y_i = k + \rho(\alpha t_i)b_1x_{1i} + \rho(\alpha t_i)b_2x_{2i} \dots + r_i,
\]</span></p>
<p>where <span class="math inline">\(x_{ij}\)</span> is the current state of predictor variable <span class="math inline">\(j\)</span> for species <span class="math inline">\(i\)</span>, and <span class="math inline">\(r_i\)</span> is the residual term. The intercept (<span class="math inline">\(k\)</span>) discussed below. The phylogenetic correction factor (<span class="math inline">\(\rho\)</span>) is another measure of the strength of inertia:</p>
<p><span class="math display">\[
\rho(\alpha t_i) = 1 - \frac{1 - e^{-\alpha t_i}}{\alpha t_i},
\]</span> where <span class="math inline">\(t_i\)</span> is the time from root to species <span class="math inline">\(i\)</span>. The correction factor varies between 0 and 1, and is identical for all species for ultrametric trees. While the optimal regression represents the relationship between predictor and the trait that we expect if adaptation is immediate, the evolutionary regression is the generalized least squares regression of a trait on the predictor, and is influenced by both adaptation and inertia. The optimal regression is accordingly an estimate of the optimal relationship between predictor and the trait totally free of constraints, while evolutionary regression is the ‘observed’ relationship. The magnitude of the difference between the optimal and the evolutionary regression (<span class="math inline">\(\rho\)</span>) is therefore informative of the relative importance of inertia and adaptation between the predictor and the response trait. The evolutionary and optimal regression will be identical if there is no inertia (<span class="math inline">\(\rho(\alpha t_i) = 1\)</span>). The optimal and evolutionary regressions are not alternative models, but rather two perspectives or aspects of the same model. In the random evironment model, we also need to estimate the rate of evolution of the environmental variable, <span class="math inline">\(\sigma_x\)</span>.</p>
<p>The intercept-only model is also useful for testing one of the assumptions of the OU-BM model; that the predictor variables evolve as a Brownian motion. The stationary variance (<span class="math inline">\(v_y = \sigma^2_y / 2\alpha\)</span>) of the joint OU-BM process is a measure of the relative influence of stochastic factors in the adaptive process relative to the primary adaptive force. Secondary stochasticity is generated by a combination of stochastic influences (such as changes in unmeasured selective forces, genetic drift, etc) on the response variable as it evolves towards the optimum or optima.</p>
</div>
<div id="residual-covariances" class="section level2">
<h2 class="hasAnchor">
<a href="#residual-covariances" class="anchor"></a>Residual covariances</h2>
<p>The residual covariances of the fixed-effect model were derived in Hansen (1997):</p>
<p><span class="math display">\[
\begin{aligned}
\text{Cov}[r_i, r_j] &amp;= v_y e^{-\alpha t_{ij}}(1 - e^{-2 \alpha t_a}),
\end{aligned}
\]</span> where <span class="math inline">\(t_i\)</span> is the time from root to species <span class="math inline">\(i\)</span>, <span class="math inline">\(t_a\)</span> is the time from root to the most recent common ancestor of species <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span>, and <span class="math inline">\(t_{ij}\)</span> is the time separating species <span class="math inline">\(i\)</span> from species <span class="math inline">\(j\)</span> (i.e. the sum of the distances from <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span> to their most recent common ancestor). For the random-effect model, Hansen <em>et al.</em> (2008) derived the residual covariances as:</p>
<p><span class="math display">\[
\begin{aligned}
\text{Cov}[r_i, r_j] &amp;= (\frac{\sigma^2_{\theta} + \sigma^2_{y} }{2\alpha}) (1 - e^{-2\alpha t_a})e^{-\alpha t_{ij}} + \\
\sigma^2_{\theta}(t_a \frac{1 - e^{-at_i}}{\alpha t_i} \frac{1 - e^{-at_j}}{\alpha t_j} - &amp;\frac{1 - e^{-at_a}}{\alpha}(e^{-\alpha t_{ia}} \frac{1 - e^{-\alpha t_j}}{\alpha t_j} + e^{-\alpha t_{ja}} \frac{1 - e^{-\alpha t_i}}{\alpha t_i}))
\end{aligned}
\]</span></p>
<p>The parameter <span class="math inline">\(\sigma^2_\theta\)</span> in the above equation equals <span class="math inline">\(b_1^2\sigma_{x1}^2 + b_2^2\sigma_{x2}^2 + \dots\)</span>, assuming the predictor variables are not correlated. The instantaneous variances, <span class="math inline">\(\sigma_{xi}^2\)</span>, of the predictor variables are estimated in advance. Note that this is considerably more complex than with fixed effects, and that the residual covariances also depend on the regression parameters <span class="math inline">\(b_i\)</span>.</p>
</div>
<div id="intercept" class="section level2">
<h2 class="hasAnchor">
<a href="#intercept" class="anchor"></a>Intercept</h2>
<p>The intercept, <span class="math inline">\(k\)</span>, in the random-effect model is given as:</p>
<p><span class="math display">\[
k = e^{-\alpha t}y_a + (1 - e^{- \alpha t})b_0 + (1 - e^{-\alpha t} - \rho(\alpha t))(b_1x_{a1} + b_2x_{a2} + \dots),
\]</span></p>
<p>where <span class="math inline">\(y_a\)</span> and <span class="math inline">\(x_{aj}\)</span> are the ancestral states of the trait, <span class="math inline">\(y\)</span>, and the predictor variables, <span class="math inline">\(x_j\)</span>, and <span class="math inline">\(b_0\)</span> is the true intercept of the optimal regression. As <span class="math inline">\(t\)</span> is the time from root to the species, we also see that the intercept on a non-ultrametric tree will differ from species to species. Clearly, <span class="math inline">\(k\)</span> is not a parameter of biological interest. What we would like is to get estimates of its components, and in particular of the “optimal” intercept, <span class="math inline">\(b_0\)</span>. For species on a non-ultrametric phylogeny, it is technically possible to get independent estimates of <span class="math inline">\(b_0, y_a\)</span> and the <span class="math inline">\(x_a\)</span>’s, but unless there is strong phylogenetic inertia and the phylogeny is strongly non-ultrametric, these would be poorly resolved, and the individual estimates will be inaccurate. We generally do not recommend separate estimates of these parameters, and for an ultrametric phylogeny, we can only obtain an estimate of the composite <span class="math inline">\(k\)</span>. An estimate of the optimal intercept may, however, be obtained indirectly with an additional assumption. First, note that SLOUCH returns independent estimates of the ancestral states <span class="math inline">\(x_a\)</span> based on the assumption that the predictors evolve as if by a Brownian motion. The problem is then only to get an estimate of the ancestral state of the trait, <span class="math inline">\(y_a\)</span>. One resonable way to do this is to assume that the common ancestor was optimal, and use <span class="math inline">\(y_a = b_0 + b_1x_{a1} + b_2x_{a2} \dots\)</span>. From this we can derive an estimate of the optimal intercept as:</p>
<p><span class="math display">\[
b_0 = k + (\rho(\alpha t) - 1)(b_1x_{a1} + b_2x_{a2} \dots),
\]</span> where <span class="math inline">\(k, b\)</span>’s, <span class="math inline">\(x_a\)</span>’s and <span class="math inline">\(\rho(\alpha t)\)</span> are all output from the SLOUCH analysis. This may be useful for plotting of the optimal regression, etc. The error due to the ancestral-optimality assumption is likely to be small unless the phylogenetic half-life is large.</p>
</div>
</div>
<div id="brownian-motion-models" class="section level1">
<h1 class="hasAnchor">
<a href="#brownian-motion-models" class="anchor"></a>Brownian-motion models</h1>
<p>The Ornstein-Uhlenbeck process is a generalization of Brownian motion. When <span class="math inline">\(\alpha = 0\)</span>, the model collapses into a Brownian motion with no pull towards the optimum. Brownian motion is not distinguishable from an Ornstein-Uhlenbeck process where the phylogenetic half-life is much larger than the tree length (<span class="math inline">\(t_{1/2} &gt;&gt; t\)</span>). SLOUCH can also explicitly fit Brownian Motion models of trait evolution. The stochastic differential equation is</p>
<p><span class="math display">\[
dy = \sigma_y dW_y,
\]</span></p>
<p>where <span class="math inline">\(dW_y \sim N(0, dt)\)</span>. For a zero-trend model, we have:</p>
<p><span class="math display">\[
y_i = y_a + r_i,
\]</span> where <span class="math inline">\(y_a\)</span> is the ancestral state (or “phylogenetic mean”). The residual covariances are</p>
<p><span class="math display">\[
\text{Cov}[r_i, r_j] = \sigma_y^2t_a,
\]</span> where <span class="math inline">\(t_a\)</span> is the sum of shared branch lengths for species <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span>. With this model, we only estimate one parameter with likelihood: <span class="math inline">\(\sigma_y^2\)</span>.</p>
<div id="regime-dependent-trends" class="section level2">
<h2 class="hasAnchor">
<a href="#regime-dependent-trends" class="anchor"></a>Regime-dependent trends</h2>
<p>Keep in mind that, since there is no directional or stabilizing trend in a standard Brownian motion, the intercept is the same as the ancestral state <span class="math inline">\(y_a\)</span>. Unlike the OU process, there are no optima (<span class="math inline">\(\theta\)</span>), and the rate of adaptation (<span class="math inline">\(\alpha\)</span>) is not a part of the model. We can, however, expand the model to include regime-dependent trends:</p>
<p><span class="math display">\[
dy = \tau dt + \sigma_y dW_y
\]</span></p>
<p>where <span class="math inline">\(\tau\)</span> is one or more trends. The trends (<span class="math inline">\(\tau\)</span>) in this model are equivalent to <span class="math inline">\(\tau = \lim_{\alpha \to 0}(\theta\alpha)\)</span> under an OU model. The model predicts: <span class="math display">\[
y_i = y_a + \sum_s{\tau_s t_s} + r_i,
\]</span> where <span class="math inline">\(t_s\)</span> is the total time that species <span class="math inline">\(i\)</span> has spent in regime <span class="math inline">\(s\)</span>. With ultrametric trees, <span class="math inline">\(y_a\)</span> can not be estimated independently of the trends <span class="math inline">\(\tau\)</span>, and as such <span class="math inline">\(y_a\)</span> is not estimated with default settings. In other words, we assume that <span class="math inline">\(y_a = 0\)</span>, and hence the trends <span class="math inline">\(\tau_s\)</span> can only be interpreted relative to each other. If we do estimate <span class="math inline">\(y_a\)</span> independently, however, the trends <span class="math inline">\(\tau_s\)</span> will represent the absolute expected direction of change. The residuals covariances are equal to the zero-trend model.</p>
</div>
<div id="continuous-covariates" class="section level2">
<h2 class="hasAnchor">
<a href="#continuous-covariates" class="anchor"></a>Continuous covariates</h2>
<p>Next, we can expand the model to include continuous covariates. If we model the covariates to influence the response trait directly and immediately, the model implementation is similar to the previously described direct-effect model. Another option is to model the trend as a linear function of a random variable evolving as a Brownian motion (<span class="math inline">\(\tau = a + bx\)</span>).</p>
<p><span class="math display">\[
\begin{aligned}
dy &amp;= (a + bx)dt + \sigma_y dW_y \\
dx &amp;= \sigma_x dW_x
\end{aligned}
\]</span> where <span class="math inline">\(a\)</span> are the regime-dependent trends as previously denoted <span class="math inline">\(\tau\)</span>, and <span class="math inline">\(x\)</span> is a random variable evolving as a Brownian motion. In the case of multiple independent predictors, we have <span class="math display">\[
dy = (a + \sum_k{b_j x_j})dt + \sigma_y dW_y,
\]</span> where <span class="math inline">\(\sum_j{b_jx_j}\)</span> is also a Brownian motion with variances <span class="math inline">\(\sigma_\theta^2 = \sum_j{b_j^2 \sigma_j^2}\)</span>. For each species, the model predicts:</p>
<p><span class="math display">\[
y = y_a + \sum_s{a_st_s} + \sum_j{\rho(t)b_jx_{j}} + r,
\]</span> where <span class="math inline">\(y_a\)</span> is the ancestral state, <span class="math inline">\(a_s\)</span> is the trend in regime <span class="math inline">\(s\)</span>, and <span class="math inline">\(t_s\)</span> is the sum of time spent in regime <span class="math inline">\(s\)</span>. <span class="math inline">\(x_{j}\)</span> is the current (extant) state of predictor variable <span class="math inline">\(j\)</span>, and <span class="math inline">\(\rho(t)\)</span> is the phylogenetic correction factor:</p>
<p><span class="math display">\[
\rho(t) = \frac{t}{2},
\]</span> where <span class="math inline">\(t\)</span> is the time from the species to the root of the tree. Note that unlike previously, where <span class="math inline">\(\rho\)</span> was unitless and in the interval [0,1], the <span class="math inline">\(\rho\)</span> for the Brownian motion continuous-trend model is positive and in units of time. The residual covariances for this model are</p>
<p><span class="math display">\[
\text{Cov}[r_i, r_j] = \sigma_y^2 t_a + \sigma_\theta^2 t_a(t_a^2/12 + t_{ia}t_{ja}/4),
\]</span></p>
<p>where the time from the most recent common ancestor (MRCA) of species <span class="math inline">\(i, j\)</span> to species <span class="math inline">\(i\)</span> is <span class="math inline">\(t_{ia}\)</span>, and the time from the MRCA to species <span class="math inline">\(j\)</span> is <span class="math inline">\(t_{ja}\)</span>.</p>
<p><strong>Comment</strong>: The brownian-motion section is a little thin. We can’t just cite thomas’ &amp; krzysztof’s notes for the interpretation and derivation, can we? Does this kind of material even fit in a manual, if not first introduced elsewhere?</p>
</div>
</div>
<div id="parameter-estimation" class="section level1">
<h1 class="hasAnchor">
<a href="#parameter-estimation" class="anchor"></a>Parameter estimation</h1>
<div id="model-formulation" class="section level2">
<h2 class="hasAnchor">
<a href="#model-formulation" class="anchor"></a>Model formulation</h2>
<p>First, we set up the model. As an example, we model log neocortex size (mm<span class="math inline">\(^2\)</span>) as having three distinct optima (browsers, grazers, mixed feeders) and adapting in response to log brain size (g), where brain size evolves as if by a Brownian motion. For each species <span class="math inline">\(i\)</span>, this is</p>
<p><span class="math display">\[
y_i = c_{0i}y_a + c_{1i}\theta_\text{browser} + c_{2i}\theta_\text{grazer} + c_{3i}\theta_\text{mixed feeder} + b \rho (t_i) x_i + r_i.
\]</span> For the moment, let’s assume that <span class="math inline">\(y_a\)</span> is equal to the optimum present at the root of the tree (<span class="math inline">\(\theta_\text{mixed feeder}\)</span>). In matrix form, the equivalent is <span class="math display">\[
\mathbf{Y} = 
\begin{bmatrix}
    c_{11} &amp; c_{21} &amp; c_{31} + c_{01} &amp; \rho (t_1) x_1 \\
    c_{12} &amp; c_{22} &amp; c_{32} + c_{02} &amp; \rho (t_2) x_2  \\
    \vdots &amp; \vdots &amp; \vdots &amp; \vdots \\
    c_{1n} &amp; c_{2n} &amp; c_{3n} + c_{0n} &amp; \rho (t_n) x_n  
\end{bmatrix}
\begin{bmatrix}
  \begin{aligned}
    &amp; \theta_\text{browser} \\
    &amp; \theta_\text{grazer} \\
    &amp; \theta_\text{mixed feeder} \\
    &amp; b
  \end{aligned}
\end{bmatrix}
+  \mathbf{r}
,
\]</span> where <span class="math inline">\(\mathbf{Y}\)</span> is a vector of observed log neocortex size, <span class="math inline">\(\mathbf{r} \sim \mathcal{N}(\mathbf{0, V})\)</span> is a vector of residuals, and <span class="math inline">\(n\)</span> is the number of species. Note that the coefficients <span class="math inline">\(c\)</span> are themselves functions of the rate of adaptation (<span class="math inline">\(\alpha\)</span>), the regime topology and the tree branch lengths. In the next section we will use the equivalent compact version: <span class="math display">\[
\mathbf{Y} = \mathbf{X} \beta + \mathbf{r},
\]</span> where <span class="math inline">\(\mathbf{X}\)</span> is the model matrix, and <span class="math inline">\(\beta\)</span> is a vector of linear model parameters.</p>
</div>
<div id="least-squares" class="section level2">
<h2 class="hasAnchor">
<a href="#least-squares" class="anchor"></a>Least-squares</h2>
<p>In SLOUCH, the regression parameters are estimated using the least-squares method. Conditional on the other model parameters (e.g. <span class="math inline">\(t_{1/2}\)</span>, <span class="math inline">\(\sigma^2_y/2\alpha\)</span> and <span class="math inline">\(\sigma^2_{xj}\)</span>), we start with an ordinary least-squares estimate of the regression parameters:</p>
<p><span class="math display">\[
\hat{\mathbf{\beta}}_0 = (\mathbf{X}^T\mathbf{X})^{-1}(\mathbf{X}^T\mathbf{X}).
\]</span> We require an initial estimate <span class="math inline">\(\hat{\mathbf{\beta}}_0\)</span> since the regression coefficients enter the residual variance-covariance matrix <span class="math inline">\(\mathbf{V}\)</span> when measurement error is included in the model. We use <span class="math inline">\(\hat{\beta}_0\)</span> and <span class="math inline">\(\mathbf{V}\)</span> to obtain a generalized least-squares (GLS) estimate of the regression parameters as:</p>
<p><span class="math display">\[
\hat{\mathbf{\beta}} = (\mathbf{X}^T\mathbf{V}^{-1}\mathbf{X})^{-1}(\mathbf{X}^T\mathbf{V}^{-1}\mathbf{X}).
\]</span> We update <span class="math inline">\(\mathbf{V}\)</span> and iterate the GLS estimation until convergence is reached. For models with random predictor variables, this estimate yields the optimal or primary regression, as the interpretation depends on the phylogenetic half-life. In calculating the evolutionary regression, <span class="math inline">\(\hat{\beta}_\text{evolutionary}\)</span>, we reconstruct the model matrix <span class="math inline">\(\mathbf{X}\)</span> by setting <span class="math inline">\(\rho = 1\)</span>, and apply same above formula. The variances and covariances of the linear model parameter are estimated as:</p>
<p><span class="math display">\[
\text{Var}[\hat{\beta}] = (\mathbf{X}^T\mathbf{V}^{-1}\mathbf{X})^{-1},
\]</span> where the square root of the diagonal elements are the standard errors.</p>
</div>
<div id="maximum-likelihood" class="section level2">
<h2 class="hasAnchor">
<a href="#maximum-likelihood" class="anchor"></a>Maximum likelihood</h2>
<p>We use the GLS estimates to we evaluate the log-likelihood (= support) function:</p>
<p><span class="math display">\[
\log \mathcal{L}(t_{1/2}, \sigma_y^2/2\alpha) = -\frac{1}{2}(n\log(2\pi) + \log \left|\mathbf{V} \right| + (\mathbf{y} - \mathbf{X\hat{\beta}})^T\mathbf{V}^{-1}(\mathbf{y} - \mathbf{X\hat{\beta}})).
\]</span></p>
<p>The regression parameters and likelihood are estimated in equal fashion for all types of models. SLOUCH includes two options for estimating <span class="math inline">\(t_{1/2}\)</span> and <span class="math inline">\(\sigma^2_y/2\alpha\)</span> using likelihood. A numerical optimization routine (L-BFGS-B, Byrd <em>et al.</em> 1995) is used as default, but a grid search is also implemented.</p>
</div>
</div>
<div id="model-selection" class="section level1">
<h1 class="hasAnchor">
<a href="#model-selection" class="anchor"></a>Model selection</h1>
<p>SLOUCH output provides the user with various statistics in order to compare models with different combinations of predictor variables. These are the coefficient of determination (<span class="math inline">\(\text{R}^2\)</span>), Akaike Information Criteion (AIC), the small-sample corrected AIC (AICc) and the stricter Schwarz Information Critetion (SIC) (Burnham &amp; Anderson, 1998). For continuous predictor models, these criteria can be used to choose the number of informative predictors in a multiple regression, or to decide which is the best predictor of a given trait’s evolution. These information criteria can also be used to decide whether a single or multiple optima Ornstein-Uhlenbeck process or a Brownian-Motion process best describes a trait’s evolution. For models with fixed categorical predictors, these information criteria are useful for assessing which categories influence a trait’s evolution (Butler &amp; King 2004). In this way the user can either pool different categories or further subdivide a given category into biological meaningful categories in order to test various hypotheses concerning differential selective niches. We note however, that the information criteria will sometimes pick models with strange parameter estimates like optima or ancestral states that are way out of biological bounds, or stationary variances of zero. Such models should be regarded with skepticism, and one can use grid search to look at the support surfaces to see if there are other peaks in the likelihood surface with more reasonable estimates. Note that all of the model fit statistics, as well as the parameter search itself, is conditional on the <em>naive</em> GLS estimator, i.e. prior to any bias-correction routines.</p>
</div>
<div id="estimation-error" class="section level1">
<h1 class="hasAnchor">
<a href="#estimation-error" class="anchor"></a>Estimation error</h1>
<p>The linear model coefficients are reported with standard errors, which can be used to construct confidence intervals. These are, however, conditional on the maximum-likelihood estimates of <span class="math inline">\(t_{1/2}\)</span> and <span class="math inline">\(v_y\)</span>.</p>
<p>The estimation errors in <span class="math inline">\(t_{1/2}\)</span> and <span class="math inline">\(v_y\)</span> are more difficult to estimate, however, as we use numerical techniques to optimize the likelihood. With numerical optimization, the default setting does not provide estimation error. An experimental option (<code><a href="../reference/slouch.fit.html">slouch.fit(...hessian=TRUE)</a></code>) uses numerical sampling to approximate the Hessian matrix at the likelihood peak, which can be used to report standard errors of <span class="math inline">\(t_{1/2}\)</span> and <span class="math inline">\(v_y\)</span>. We note that this method can be numerically unstable, particularly with long half-lives, and may even crash the program for small half-lives.</p>
<p>A good way to inspect estimation error is to perform a grid search, which can be seen in the examples vignette. A grid search starts by manually providing vectors of potential values for each parameter to the program to find the combination that maximizes the likelihood. The best estimates of <span class="math inline">\(t_{1/2}\)</span> and <span class="math inline">\(v_y\)</span> are a joint estimate and for this reason the support region (the three-dimensional log-likelihood plot) is given as a measure of uncertainty for <span class="math inline">\(t_{1/2}\)</span> and <span class="math inline">\(v_y\)</span> jointly. The marginal lowest and highest values of <span class="math inline">\(t_{1/2}\)</span> within the support region, conditional on the best estimate of <span class="math inline">\(v_y\)</span>, can also be obtained using the figure and the support-region tables in the output (and vice versa for <span class="math inline">\(v_y\)</span>). The <code>slouch.fit</code> function can also take scalars rather than vectors, which may be useful for finding the marginal support region of <span class="math inline">\(t_{1/2}\)</span> conditional on <span class="math inline">\(v_y\)</span> (and vice versa).</p>
<div id="measurement-variance" class="section level2">
<h2 class="hasAnchor">
<a href="#measurement-variance" class="anchor"></a>Measurement variance</h2>
<p>Hansen &amp; Bartozsek (2012) discuss how to incorporate measurement error into a phylogenetic comparative analysis. Most often, a comparative analysis involves using species means as input values. In this case the main component of measurement error is estimation error in the mean (or whatever other statistics are used) resulting from sampling a finite number of individuals. If each mean species value to be used in the regression was obtained from a large number of individuals (&gt; 20-30 or so) the square of the standard error (i.e. the estimation variance) can be entered into the measurement error column for that variable. If no measurement error is to be included, then simply add a column of zeros. We also refer the reader to the examples vignette for a more detailed example. SLOUCH deals with measurement error in the response variable by adding measurement variance to the diagonals of the residual variance matrix. Measurement variance in the random predictor variable(s) is incorporated by multiplying it by the square of the uncorrected slope parameter and then adding it to the diagonals of the residual variance matrix. If the observational errors in the predictor variables are non-zero, the generalized least squares estimator may be biased. A bias correction for the regression coefficients is implemented according to Hansen &amp; Bartoszek (2012).</p>
</div>
</div>
<div id="caveats-and-limitations" class="section level1">
<h1 class="hasAnchor">
<a href="#caveats-and-limitations" class="anchor"></a>Caveats and limitations</h1>
<p>As with any statistical model, the quality of the parameter estimates is dependent on the quality of the data and certain assumptions concerning the residuals must be met. SLOUCH utilizes both measured trait values and the phylogenetic topology and branch lengths as data. With regards to the trait values, we assume that a linear relationship exists between expected optimal trait values and the predictor variables. Although SLOUCH can perform non-linear regression by entering quadratic terms in the model, this will most likely violate the assumption that the predictors evolve as a Brownian Motion. SLOUCH also assumes that the underlying evolutionary process is equal in all parts of the tree, as <span class="math inline">\(t_{1/2}\)</span> and <span class="math inline">\(\sigma_y^2\)</span> are constant. When studying a clade that is heterogeneic (e.g. Cetartiodactyla) or spans large timespans this assumption may be violated, and one may consider splitting the phylogeny or using methods that relax this assumption (e.g. Beaulieu <em>et al</em>. 2014). It should also be kept in mind that any inferences drawn about the rates of adaption, phylogenetic inertia and its potential effects on trait values are conditional on the phylogeny used. SLOUCH does not include routines for evaluating the effects of phylogeny uncertainty in a systematic manner, but one can perform the analysis on plausible alternative phylogenetic hypotheses to examine the effects of phylogenetic uncertainty on parameter estimates. Simulation may also be useful here for examining potential interactions between true parameter values and alternative phylogenies.</p>
</div>
<div id="history-comparison-with-other-packages" class="section level1">
<h1 class="hasAnchor">
<a href="#history-comparison-with-other-packages" class="anchor"></a>History &amp; comparison with other packages</h1>
<p>This document describes an R-package for estimating parameters for a suite of comparative methods based on the Ornstein-Uhlenbeck process as a model of adaptive evolution. We refer to this program as SLOUCH 2.0 since it builds on the models in SLOUCH 1.0 (Hansen <em>et al.</em> 2008). SLOUCH 1.0 dealt with a continuous trait that evolves as an Ornstein-Uhlenbeck process around an optimal state that itself depend on an environmental variable that evolves as if by a Brownian motion. For the current version we have extended the model to deal with an optimal state that depends on multiple randomly evolving predictors and multiple direct effect predictors. We also describe in more detail how to include measurement error in the analyses. SLOUCH 2.0 also includes functionality designed to deal with traits that evolve in response to fixed categorical niches that can be mapped onto a phylogeny and then used to estimate niche optima for the response trait (Hansen, 1997; Butler &amp; King, 2004). The multi-optima model is thus an alternative to Butler &amp; King’s (2004) OUCH program, but it differs in a few crucial apects, as described below. SLOUCH 1.3 and earlier versions were built as an addition to OUCH, and data formats and some functions were borrowed from the original version of OUCH, either as is, or slightly modified. In SLOUCH 2.0 the phylogenetic tree format is revamped, and now uses the format used by the R-package APE (Paradis <em>et al.</em> 2004).</p>
<p>The fixed.fit function of SLOUCH 1.1 differs from the OUCH program as follows: a) it allows the user to estimate parameters using a non-ultrametric tree, such as those that include extinct species, b) it allows the user to incorporate measurement error in the response variable, c) it automatically calculates standard errors for estimated optima, goodness of fit as <span class="math inline">\(\text{R}^2\)</span> and can be used to compute support regions for rates of adaptation and trait variances, d) it includes functionality to use both grid search and the built-in R optimization routines to estimate parameters, and e) it includes various options to deal with intercept estimation, which can be problematic. [Do we need to compare with OUCH?]</p>
<p>SLOUCH 2.0 also includes Brownian-motion models where the optima are instead modeled as trends. This document is written to serve as an updated user manual for SLOUCH. It provides a summary of the rationale for the methods, and illustrates them with examples based on the ruminant neocortex data set. It extends the user manual for SLOUCH 1.0 by discussing multiple regression and ANOVA type of models for the optimum, and gives more details on modeling non-ultrametric trees and measurement error. The mathematical and conceptual basis of the adaptation-inertia methods are presented and discussed in Hansen (1997), Butler &amp; King (2004), and Hansen et al. (2008). A user of SLOUCH should also consult these papers.</p>

</div>
<div id="references" class="section level1">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<p>Burnham, K. P. &amp; Anderson, D. R. (1998). Model selection and inference: A practical information- theoretic approach. Springer.</p>
<p>Butler, M. A., &amp; King, A. A. (2004). Phylogenetic comparative analysis: a modeling approach for adaptive evolution. American Naturalist, 164(6), 683–695. <a href="https://doi.org/10.1086/426002" class="uri">https://doi.org/10.1086/426002</a></p>
<p>Edwards, A. W. F. (1992). Likelihood. expanded edition Johns Hopkins University Press. Baltimore, MD.</p>
<p>Escudero, M., Hipp, A. L., Hansen, T. F., Voje, K. L., &amp; Luceño, M. (2012). Selection and inertia in the evolution of holocentric chromosomes in sedges (Carex, Cyperaceae). New Phytologist, 195(1), 237–247. <a href="https://doi.org/10.1111/j.1469-8137.2012.04137.x" class="uri">https://doi.org/10.1111/j.1469-8137.2012.04137.x</a></p>
<p>Grabowski, M., Voje, K. L., &amp; Hansen, T. F. (2016). Evolutionary modeling and correcting for observation error support a 3/5 brain-body allometry for primates. Journal of human evolution, 94, 106-116.</p>
<p>Hansen, T. F. (1997). Stabilizing Selection and the Comparative Analysis of Adaptation. Evolution, 51(5), 1341. <a href="https://doi.org/10.2307/2411186" class="uri">https://doi.org/10.2307/2411186</a></p>
<p>Hansen, T. F., &amp; Martins, E. P. (1996). Translating between microevolutionary process and macroevolutionary patterns: the correlation structure of interspecific data. Evolution, 50(4), 1404-1417.</p>
<p>Hansen, T. F., &amp; Orzack, S. H. (2005). Assessing current adaptation and phylogenetic inertia as explanations of trait evolution: the need for controlled comparisons. Evolution, 59(10), 2063-2072.</p>
<p>Hansen, T. F., Pienaar, J., &amp; Orzack, S. H. (2008). A comparative method for studying adaptation to a randomly evolving environment. Evolution, 62(8), 1965–1977. <a href="https://doi.org/10.1111/j.1558-5646.2008.00412.x" class="uri">https://doi.org/10.1111/j.1558-5646.2008.00412.x</a></p>
<p>Hansen, T. F., &amp; Bartoszek, K. (2012). Interpreting the evolutionary regression: The interplay between observational and biological errors in phylogenetic comparative studies. Systematic Biology, 61(3), 413–425. <a href="https://doi.org/10.1093/sysbio/syr122" class="uri">https://doi.org/10.1093/sysbio/syr122</a></p>
<p>Hansen, T. F. (2014). Use and misuse of comparative methods in the study of adaptation. In Garamszegi &amp; L. Zsolt (Eds.), Modern Phylogenetic Comparative Methods and their Application in Evolutionary Biology (pp. 351–379). Springer. <a href="https://doi.org/10.1007/978-3-662-43550-2_14" class="uri">https://doi.org/10.1007/978-3-662-43550-2_14</a></p>
<p>Harvey, P. H., &amp; Pagel, M. D. (1991). The comparative method in evolutionary biology (Vol. 239). Oxford: Oxford university press.</p>
<p>Martins, E. P., &amp; Hansen, T. F. (1997). Phylogenies and the comparative method: a general approach to incorporating phylogenetic information into the analysis of interspecific data. The American Naturalist, 149(4), 646-667.</p>
<p>O’Meara, B. C., &amp; Beaulieu, J. M. (2014). Modelling stabilizing selection: The attraction of Ornstein-Uhlenbeck models. In Garamszegi &amp; L. Zsolt (Eds.), Modern Phylogenetic Comparative Methods and their Application in Evolutionary Biology (pp. 381–393). Springer. <a href="https://doi.org/10.1007/978-3-662-43550-2_15" class="uri">https://doi.org/10.1007/978-3-662-43550-2_15</a></p>
<p>Labra, A., Pienaar, J., &amp; Hansen, T. F. (2009). Evolution of Thermal Physiology in Liolaemus Lizards: Adaptation, Phylogenetic Inertia, and Niche Tracking. The American Naturalist, 174(2), 204–220. <a href="https://doi.org/10.1086/600088" class="uri">https://doi.org/10.1086/600088</a></p>
<p>Ridley, M. (1983). The explanation of organic diversity: the comparative method and adaptations for mating. Oxford University Press, USA.</p>
<p>Toljagić, O., Voje, K. L., Matschiner, M., Liow, L. H., &amp; Hansen, T. F. (2017). Millions of years behind: Slow adaptation of ruminants to grasslands. Systematic Biology, (318). <a href="https://doi.org/10.1093/sysbio/syx059" class="uri">https://doi.org/10.1093/sysbio/syx059</a></p>
<p>Sober, E. (2008). Evidence and evolution: The logic behind the science. Cambridge University Press.</p>
<p>Beaulieu, J. M., Jhwueng, D. C., Boettiger, C., &amp; O’Meara, B. C. (2012). Modeling stabilizing selection: expanding the Ornstein–Uhlenbeck model of adaptive evolution. Evolution, 66(8), 2369-2383.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#comparative-methods-for-adaptive-hypotheses">Comparative methods for adaptive hypotheses</a><ul class="nav nav-pills nav-stacked">
<li><a href="#background">Background</a></li>
      <li><a href="#phylogeny">Phylogeny</a></li>
      </ul>
</li>
      <li>
<a href="#comparative-methods-built-around-an-ornstein-uhlenbeck-process">Comparative methods built around an Ornstein-Uhlenbeck process</a><ul class="nav nav-pills nav-stacked">
<li><a href="#the-optimum">The optimum</a></li>
      <li><a href="#ornstein-uhlenbeck-process">Ornstein-Uhlenbeck process</a></li>
      <li><a href="#parameterization">Parameterization</a></li>
      <li><a href="#random-continuous-predictor">Random continuous predictor</a></li>
      <li><a href="#categorical-fixed-predictors-niches-regimes">Categorical fixed predictors (niches, regimes)</a></li>
      <li><a href="#direct-effect-predictors">Direct-effect predictors</a></li>
      <li><a href="#example-expectation-with-random-covariates">Example: expectation with random covariates</a></li>
      <li><a href="#residual-covariances">Residual covariances</a></li>
      <li><a href="#intercept">Intercept</a></li>
      </ul>
</li>
      <li>
<a href="#brownian-motion-models">Brownian-motion models</a><ul class="nav nav-pills nav-stacked">
<li><a href="#regime-dependent-trends">Regime-dependent trends</a></li>
      <li><a href="#continuous-covariates">Continuous covariates</a></li>
      </ul>
</li>
      <li>
<a href="#parameter-estimation">Parameter estimation</a><ul class="nav nav-pills nav-stacked">
<li><a href="#model-formulation">Model formulation</a></li>
      <li><a href="#least-squares">Least-squares</a></li>
      <li><a href="#maximum-likelihood">Maximum likelihood</a></li>
      </ul>
</li>
      <li><a href="#model-selection">Model selection</a></li>
      <li>
<a href="#estimation-error">Estimation error</a><ul class="nav nav-pills nav-stacked">
<li><a href="#measurement-variance">Measurement variance</a></li>
      </ul>
</li>
      <li><a href="#caveats-and-limitations">Caveats and limitations</a></li>
      <li><a href="#history-comparison-with-other-packages">History &amp; comparison with other packages</a></li>
      <li><a href="#references">References</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Bjørn Tore Kopperud, Jason Pienaar, Kjetil Lysne Voje, Steven Hecht Orzack, Thomas F. Hansen.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
